### PR TITLE
API-32089: Validate 2122 V2

### PR DESCRIPTION
**Pull request by dellerbie**
## Summary

Creatively borrowed from [2122a validator](https://github.com/department-of-veterans-affairs/vets-api/pull/15131)

- Adds Claims API V2 JSON schema validation for form 21-22 to appoint an organization as POA
- Uses the same schema as the V1 2122 form

## Related issue(s)
- [API-32089](https://jira.devops.va.gov/browse/API-32089)

## Testing done

Tested the following scenarios in postman locally:
- invalid ccg token throws 401
- request body is not JSON throws 422
- ICN is not real throws 404
- the request body does not pass JSON validation throws 422
- no representatives have the POA code throws 404
- when everything is right (token, ICN, JSON body, and POA code) get 200

**Invalid CCG Token**
image

**Request body is not JSON throws 422**
image

 **ICN is not real throws 404**
image

**Request body does not pass JSON validation throws 422**
image

**No representatives have the POA code throws 404**
image

**When everything is right (token, ICN, JSON body, and POA code) get 200**
image

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney_controller.rb [hunk 1]
```diff
@@ -24,18 +24,38 @@ def show
...
 
           submit_power_of_attorney(poa_code)
         end
 
+        def validate2122
+          target_veteran
+          validate_json_schema
+
+          poa_code = form_attributes.dig('serviceOrganization', 'poaCode')
+          validate_individual_poa_code!(poa_code)
```
[the remark this answers is not in the document]
Comment (stiehlrod) [reply modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney_controller.rb:43]: Good catch! 
```irb(main):009:0> poa_code = "1NY"
=> "1NY"
irb(main):010:0> ::Veteran::Service::Organization.where('? = poa', poa_code).any?
2024-01-29 10:39:43.935953 D [76713:12360 log_subscriber.rb:130] ActiveRecord::Base --   Veteran::Service::Organization Exists? (1.6ms)  SELECT 1 AS one FROM "veteran_organizations" WHERE ('1NY' = poa) LIMIT $1  [["LIMIT", 1]]
=> true  ```

[the remark this answers is not in the document]
Comment (jvcAdHoc) [reply modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney_controller.rb:43]: V1 validation endpoint doesn't make the distinction between 2122 and 2122a so as long as the poa code is in the reps table then it's "good".

I think using the organization table makes sense here now that there's a separate 2122a endpoint.

File: modules/claims_api/config/schemas/v2/2122a.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "$schema": "http://json-schema.org/draft-07/schema#",
-  "description": "Form 2122 Schema",
+  "description": "Form 2122a Schema",
```
[the remark this answers is not in the document]
Comment (jvcAdHoc) [reply modules/claims_api/config/schemas/v2/2122a.json:3]: That's a good catch, thanks!

**Review by mchristiansonVA — DISMISSED** [review]
Tested branch locally via Postman and confirmed correct response messages are displayed for various error and normal cases. Rspec tests added seem to cover the necessary cases; ran those locally as well without issues. Looks good to me!

**Review by stiehlrod — APPROVED** [review]

**Pull request merged**